src/main/java/com/uber/cadence/internal/replay/ReplayDeciderCache.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.internal.replay;
+
+import com.uber.cadence.internal.common.LruCache;
+
+class ReplayDeciderCache extends LruCache<ReplayDecider> {
```

Comment:
Don't use inheritance to configure. Just instantiate LruCache with appropriate constructor parameters.